src/instrument.cpp

```diff
@@ -282,15 +286,82 @@ void BytecodeRewriter::rewriteCode() {
     put16(max_locals);
 
     u32 code_length = get32();
-    put32(code_length + EXTRA_BYTECODES);
+    const u8* code = get(code_length);
+    u32 code_length_idx = _dst_len;
+    put32(code_length); // to be fixed later
+
+    int code_segment_begin = _dst_len;
+
+    u32 max_relocation = 0;
+    for (u32 i = 0; i < code_length;) {
+        u8 opcode = code[i];
+
+        if (opcode == 0xb0 || opcode == 0xaf || opcode == 0xae || opcode == 0xac ||
+            opcode == 0xad || opcode == 0xb1 || opcode == 0xbf) {
+            max_relocation += EXTRA_BYTECODES;
+        } else if (opcode == 0xa8 || opcode == 0xa7) {
+            max_relocation += 2;
+        }
+        i += OPCODE_LENGTH[opcode];
+    }
```

Comment:
Check that `code_length + max_relocation` does not exceed maximum allowed code length.
It should be extremely rare situation - you can simply give up rewriting in this case.

Reply:
cb2d7de6aca55ac4b9b83794a489a42bfd7a9817